bundles/fr.adaussy.permadeler.model.design.tests/src/fr/adaussy/permadeler/model/design/services/ValidationServiceTest.java

```diff
@@ -0,0 +1,211 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Arthur Daussy.
+ *
+ * This program and the accompanying materials are made 
+ * available under the terms of the Eclipse Public License 2.0 
+ * which is available at https://www.eclipse.org/legal/epl-2.0/ 
+ * Contributors:
+ * Arthur Daussy - initial API and implementation.
+ ******************************************************************************/
+
+package fr.adaussy.permadeler.model.design.services;
+
+import static org.junit.Assert.assertTrue;
+
+import java.util.Arrays;
+import java.util.stream.IntStream;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import fr.adaussy.permadeler.model.Permadeler.Area;
+import fr.adaussy.permadeler.model.Permadeler.CompatibilityLink;
+import fr.adaussy.permadeler.model.Permadeler.CompatibilityMatrix;
+import fr.adaussy.permadeler.model.Permadeler.Family;
+import fr.adaussy.permadeler.model.Permadeler.Genus;
+import fr.adaussy.permadeler.model.Permadeler.KnowledgeBase;
+import fr.adaussy.permadeler.model.Permadeler.PermadelerFactory;
+import fr.adaussy.permadeler.model.Permadeler.Plantation;
+import fr.adaussy.permadeler.model.Permadeler.Planting;
+import fr.adaussy.permadeler.model.Permadeler.Root;
+import fr.adaussy.permadeler.model.Permadeler.Species;
+import fr.adaussy.permadeler.model.Permadeler.Zone;
+
+/**
+ * Tests for the service {@link ValidationService#validatePlantationCompatibility(Area)}
```

Comment:
No this class whill hold all test on all method in ValidationService (not only validatePlantationCompatibility)

Reply:
ok, it make sense. But then, should we do also some kind of class hierarchy for unit tests ?
like : 
- public class ValidationServiceTest
     - public class ValidatePlantationCompatibilty
          - test1
          - test 2...
     - public class WhateverServiceWillComeAfter
     - private static class InputModelHelper

Reply:
What we usually do it. On test class for one service class. All test methods are public method anotated with @test. No no need for sub inner class. The name of the method could be used to identify the method under test : checkValidatePlantationCompatibiltyWith<Context>
For example : checkValidatePlantationCompatibiltyWith2CompatGenuses

Reply:
ok, i changed the names so it's clearer